### PR TITLE
Add Grid Schema Smart contract  

### PR DESCRIPTION
**Pull request by agunde406**
This PR implements the Smart contract described in
the Grid Primitives RFC
hyperledger/grid-rfcs#4

Note this does not include the permission check of if
the agent has the permission can_create_schema for the
organization. This we will be added in a separate PR
after the implementation of Permission checker in the
Grid SDK.

### CODE REVIEW

File: contracts/schema/src/payload.rs [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+cfg_if! {
+    if #[cfg(target_arch = "wasm32")] {
+        use sabre_sdk::ApplyError;
+    } else if #[cfg(test)]{
+        use sawtooth_sdk::processor::handler::ApplyError;
```
Comment (peterschwarz) [inline contracts/schema/src/payload.rs]: Is this different from the else branch?  I'm not seeing it, if so

Comment (agunde406) [reply contracts/schema/src/payload.rs]: Whoops, that whole test section should be removed. This is left over from before the updated sabre.

Comment (agunde406) [reply contracts/schema/src/payload.rs]: Fixed

File: contracts/schema/src/state.rs [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+                        return Ok(Some(schema.clone()));
+                    }
+                }
+                Ok(None)
+            }
+            None => Ok(None),
+        }
+    }
+
+    /// Sets a Grid Schema in state. Handles creating a SchemaList if one does not already exists
```
Comment (jsmitchell) [inline contracts/schema/src/state.rs]: s/exists/exist/

File: contracts/schema/src/state.rs [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+    fn test_get_organization_none() {
+        let mut transaction_context = MockTransactionContext::default();
+        let mut state = GridSchemaState::new(&mut transaction_context);
+
+        let result = state.get_organization("test_org").unwrap();
+        assert!(result.is_none())
+    }
+
+    #[test]
+    // Test that if an org does exists in state, Some(Organization) is returned;
```
Comment (jsmitchell) [inline contracts/schema/src/state.rs]: trailing semicolon

File: contracts/schema/src/state.rs [hunk 4]
```diff
@@ -0,0 +1,419 @@
...
+        let result = state.get_organization("test_org").unwrap();
+        assert!(result.is_some());
+        let organization = result.unwrap();
+
+        assert_eq!(organization.org_id(), "test_org");
+    }
+
+    #[test]
+    // 1. Test that if a schema is not in state a None is returned.
+    // 2. Test that a schema can be addec to state using set_state.
```
Comment (jsmitchell) [inline contracts/schema/src/state.rs]: s/addec/added/

File: contracts/schema/src/state.rs (hunk 2 above)
Comment (agunde406) [reply contracts/schema/src/state.rs]: updated

File: contracts/schema/src/state.rs (hunk 3 above)
Comment (agunde406) [reply contracts/schema/src/state.rs]: updated

File: contracts/schema/src/state.rs (hunk 4 above)
Comment (agunde406) [reply contracts/schema/src/state.rs]: updated

**Pull request merged**